Realm/ObjectServerTests/SwiftObjectServerTests.swift

```diff
@@ -2912,4 +2912,188 @@ class CombineObjectServerTests: SwiftSyncTestCase {
+        // it should not download the latest dataset
+        enum OpenKind: Int {
+            case first, second
+        }
+        switch ProcessKind.current {
+        case .parent:
+            let user = try await self.app.login(credentials: basicCredentials())
+            let user1Realm = try await Realm(configuration: user.configuration(testName: #function))
+            try! user1Realm.write {
+                user1Realm.add(SwiftHugeSyncObject.create())
+                user1Realm.add(SwiftHugeSyncObject.create())
+            }
+            waitForUploads(for: user1Realm)
+
+            let (email2, password2) = (randomString(10), "password")
+            self.runChildAndWait(with: ChildProcessEnvironment(appIds: self.appIds,
+                                                               email: email2,
+                                                               password: password2,
+                                                               identifer: OpenKind.first.rawValue,
+                                                               shouldCleanUpOnTermination: false))
+            try! user1Realm.write {
+                user1Realm.add(SwiftHugeSyncObject.create())
+                user1Realm.add(SwiftHugeSyncObject.create())
+            }
+            waitForUploads(for: user1Realm)
+
+            self.runChildAndWait(with: ChildProcessEnvironment(appIds: self.appIds,
+                                                               email: email2,
+                                                               password: password2,
+                                                               identifer: OpenKind.second.rawValue,
+                                                               shouldCleanUpOnTermination: true))
+        case .child(let environment):
+            switch OpenKind(rawValue: environment.identifier)! {
+            case .first:
+                let user = try await app
+                    .login(credentials: .emailPassword(email: environment.email!,
+                                                       password: environment.password!))
+                let realm = try await Realm(configuration: user.configuration(testName: #function),
+                                            downloadBeforeOpen: .once)
+                XCTAssertEqual(realm.objects(SwiftHugeSyncObject.self).count, 2)
+            case .second:
+                let user = try await app
+                    .login(credentials: .emailPassword(email: environment.email!,
+                                                       password: environment.password!))
+                XCTAssertTrue(Realm.fileExists(for: user.configuration(testName: #function)))
+                let realm = try await Realm(configuration: user.configuration(testName: #function),
+                                            downloadBeforeOpen: .once)
+                XCTAssertEqual(realm.objects(SwiftHugeSyncObject.self).count, 2)
```

Comment:
It looks like the only thing that changes between these two is the `XCTAssertTrue(Realm.fileExists(for: user.configuration(testName: #function)))` and duplicating the rest just makes it less clear.